rn/Teacher/src/modules/inbox/__tests__/Inbox.test.js

```diff
@@ -53,258 +54,302 @@ let defaultProps = {
+    }
+    handleRefresh(props)
+    expect(props.refreshInboxArchived).toHaveBeenCalled()
+  })
 
-it('updates on scope change', () => {
-  const tree = renderer.create(
-    <Inbox conversations={[]} scope='all' />
-  )
-
-  let props = {
-    scope: 'unread',
-    refreshInboxUnread: jest.fn(),
-    refreshCourses: jest.fn(),
-  }
-
-  setProps(tree, props)
-  expect(props.refreshInboxUnread).toHaveBeenCalled()
-
-  props = {
-    scope: 'unread',
-    refreshInboxUnread: jest.fn(),
-    refreshCourses: jest.fn(),
-  }
-
-  setProps(tree, props)
-  expect(props.refreshInboxUnread).not.toHaveBeenCalled()
-  expect(props.refreshCourses).not.toHaveBeenCalled()
-})
+  it('does not refresh courses if present', () => {
+    const props = {
+      refreshInboxArchived: jest.fn(),
+      scope: 'archived',
+      refreshCourses: jest.fn(),
+      courses: [template.course()],
+    }
+    handleRefresh(props)
+    expect(props.refreshCourses).not.toHaveBeenCalled()
+  })
 
-it('refreshed component', () => {
-  const props = {
-    conversations: [],
-    refreshCourses: jest.fn(),
-    refreshInboxAll: jest.fn(),
-    scope: 'all',
-    courses: [template.course()],
-  }
-
-  const tree = renderer.create(
-    <Refreshed {...props} />
-  )
-  expect(tree.toJSON()).toMatchSnapshot()
-  tree.getInstance().refresh()
-  setProps(tree, props)
-  expect(props.refreshCourses).not.toHaveBeenCalled()
-  expect(props.refreshInboxAll).toHaveBeenCalled()
-})
+  it('filters conversations based on course', () => {
+    const courses = [
+      { id: '1' },
+      { id: '2' },
+    ]
+    const tree = renderer.create(
+      <Inbox {...defaultProps} courses={courses} scope='all' />
+    )
+    const instance = tree.getInstance()
+    instance.setState({ selectedCourse: '1' })
+    const node = tree.toJSON()
+    expect(node).toMatchSnapshot()
+  })
 
-it('should call the right functions in handleRefresh', () => {
-  let props = {
-    refreshInboxAll: jest.fn(),
-    scope: 'all',
-    refreshCourses: jest.fn(),
-    courses: [],
-  }
-  handleRefresh(props)
-  expect(props.refreshInboxAll).toHaveBeenCalled()
-  expect(props.refreshCourses).toHaveBeenCalled()
-
-  props = {
-    refreshInboxUnread: jest.fn(),
-    scope: 'unread',
-    refreshCourses: jest.fn(),
-  }
-  handleRefresh(props)
-  expect(props.refreshInboxUnread).toHaveBeenCalled()
-
-  props = {
-    refreshInboxStarred: jest.fn(),
-    scope: 'starred',
-    refreshCourses: jest.fn(),
-  }
-  handleRefresh(props)
-  expect(props.refreshInboxStarred).toHaveBeenCalled()
-
-  props = {
-    refreshInboxSent: jest.fn(),
-    scope: 'sent',
-    refreshCourses: jest.fn(),
-  }
-  handleRefresh(props)
-  expect(props.refreshInboxSent).toHaveBeenCalled()
-
-  props = {
-    refreshInboxArchived: jest.fn(),
-    scope: 'archived',
-    refreshCourses: jest.fn(),
-  }
-  handleRefresh(props)
-  expect(props.refreshInboxArchived).toHaveBeenCalled()
-})
+  it('shows CourseFilter', () => {
```

Comment:
This is one of the two new tests I added. Everything else was moved to be nested within `describe('Inbox', () => {`